contracts/test/Proofs.t.sol

```diff
@@ -34,23 +34,33 @@ contract MerkleProofTest is Test {
 
         for (uint i = 0; i < leaves.length; i++) {
             bytes32[] memory proof = buildProof(tree, i);
-            bool result = MerkleProof.verify(proof, root, leaves[i]);
-            assertEq(result, true, string.concat("Invalid proof for leaf ", vm.toString(i)));
+            assertTrue(MerkleProof.verify(proof, root, leaves[i], i), string.concat("Invalid proof ", vm.toString(i)));
+            // Ensure the proof is invalid for every other index within range
+            for (uint j = 0; j < leaves.length; j++) {
+                if (j != i) {
+                    assertFalse(MerkleProof.verify(proof, root, leaves[i], j));
+                }
+            }
         }
     }
 
 
     function testVerifyTreesManyLeaves() public view {
-        for (uint256 width = 1; width < 100; width++) {
+        for (uint256 width = 4; width < 60; width++) {
```

Comment:
Why this change?

Reply:
Oh because you did 2 and 3 above? And 100 was overkill ?

Reply:
Re `4`, yes. Re `100`, for some reason I don't fully understand, adding the "Ensure the proof is invalid for every other index" loop caused the test to consume more memory (not just quadratic compute) and fail. I tried a few tricks to release the memory, but without luck. Even 60 is probably way more than we need.

Reply:
woah surprising